frontend/src/app/shared/utils/option.ts

```diff
@@ -9,7 +9,17 @@ export class Option {
   static of<OptionValue extends PrimitiveOption>(value: OptionValue, displayValue?: string): SelectOption<OptionValue> {
     return {
       value,
-      displayValue: displayValue || (value as string),
+      displayValue: displayValue || String(value),
     };
   }
 }
+
+export class Options {
+  static fromMap<OptionValue>(convertedMap: Map<OptionValue, string>) {
```

Comment:
As mentioned in the previous comment, it depends on how strict type information we want to have. For full generic benefit, but limited to PrimitiveOption, we should actually have:
```
 static fromMap<OptionValue extends PrimitiveOption>(convertedMap: Map<OptionValue, string>)
```
and also fix the type definitions on `toOption` method.

Then, for the following set of data:
```
const boolMap = new Map([
  [true, 'test'],
  [false, 'bb'],
]);

const stringMap = new Map([
  ['sss', 'test'],
  ['aa', 'bb'],
]);
```
we would have:
```
Options.fromMap(boolMap); // SelectOption<boolean>[]
Options.fromMap(stringMap); // SelectOption<string>[]
```

but when dropping generics, we would only have:
```
Options.fromMap(boolMap); // SelectOption<PrimitiveOption>[]
Options.fromMap(stringMap); // SelectOption<PrimitiveOption>[]
```

Reply:
It is up to discussion whether we need the precise type information with generics, however I'd personally go for the full generic solution, since we have it mid-way through now and it's fun ;)